core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1691,6 +1710,30 @@ class Partition(val topicPartition: TopicPartition,
     updatedState
   }
 
+  private def addBrokerEpochToIsr(isr: List[Int]): List[BrokerState] = {
+    isr.map { brokerId =>
+      val brokerState = new BrokerState().setBrokerId(brokerId)
+      if (!metadataCache.isInstanceOf[KRaftMetadataCache]) {
+        brokerState.setBrokerEpoch(-1)
+      } else if (brokerId == localBrokerId) {
+        brokerState.setBrokerEpoch(localBrokerEpochSupplier())
+      } else if (!remoteReplicasMap.contains(brokerId) || !remoteReplicasMap.get(brokerId).stateSnapshot.brokerEpoch.isDefined) {
```

Comment:
remoteReplicasMap can change btw the call of `contains` and `get`. So, it's probably better to save the `get` result to a local val and then do the check on the val. Ditto for `brokerEpoch`.